pages/deliverer/incoming-orders/[id].tsx

```diff
@@ -0,0 +1,201 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+import { supabase } from '../../../api'
+import { useRouter } from 'next/router'
+import { useAlert } from '../../../helpers/alertContext'
+import { supabaseClient, withPageAuth } from '@supabase/auth-helpers-nextjs'
+
+type IncomingOrderDto = {
+  id: number
+  restaurants: RestaurantDto
+  order_text: string
+  deliverer_id: string
+  ordered_at: Date
+  is_active: boolean
+  destinations: DestinationDto
+  cost: number
+}
+
+type RestaurantDto = {
+  id: number
+  name: string
+}
+
+type DestinationDto = {
+  id: number
+  name: string
+}
```

Comment:
we should expand out the types to contain all their properties, and move their definition into a separate folder to be reused in other components

Reply:
created types folder in root directory, and also a baseUrl property in tsconfig.json so we can refer to the root directory without relative paths